### PR TITLE
Add build date and revision

### PR DESCRIPTION
**Pull request by OskarDamkjaer**
When we switch to deploying more often in aura, it makes sense to be able to see when a version was built and what exact git revision it was as you can no longer rely only on the version number.

demo at http://add_build_date_and_revision.surge.sh

### CODE REVIEW

File: build_scripts/generate-manifest-helpers.js [hunk 1]
```diff
@@ -37,6 +37,33 @@ function writeDataToFile(file, data) {
...
+      .split(/.*[: ]/)
+      .slice(-1)[0]
+    if (rev.indexOf('/') === -1) {
+      return rev
+    } else {
+      return fs
+        .readFileSync('.git/' + rev)
+        .toString()
+        .trim()
+    }
```
Comment (eijawerner) [inline build_scripts/generate-manifest-helpers.js:59]: Please add a comment or link to inspiration for this and remove support for detached head which I don't think we need, to simplify things.

File: src/shared/modules/app/appDuck.ts [hunk 2]
```diff
@@ -46,6 +47,10 @@ export const inWebEnv = (state: any) => getEnv(state) === WEB
 export const inCloudEnv = (state: any) => getEnv(state) === CLOUD
 export const inWebBrowser = (state: any) => [WEB, CLOUD].includes(getEnv(state))
 export const inDesktop = (state: any) => getEnv(state) === DESKTOP
+export const getGitRevision = (state: any): string | null =>
```
Comment (eijawerner) [inline src/shared/modules/app/appDuck.ts:50]: Add GlobalState type?

File: src/shared/modules/app/appDuck.ts [hunk 3]
```diff
@@ -46,6 +47,10 @@ export const inWebEnv = (state: any) => getEnv(state) === WEB
...
+  state[NAME].gitRevision ?? null
+export const getBuiltAt = (state: any): string | null =>
```
Comment (eijawerner) [inline src/shared/modules/app/appDuck.ts:52]: same

**Review by eijawerner — COMMENTED** [review]
Nice! Added some comments.

File: src/shared/modules/app/appDuck.ts (hunk 2 above)
Comment (OskarDamkjaer) [reply src/shared/modules/app/appDuck.ts:50]: App wasn't added properly on the global state, but I went ahead and added it now :+1:

**Review by eijawerner — APPROVED** [review]
⭐ Nice to have this info in the future!

**Pull request merged**